pkg/throttle/writer.go

```diff
@@ -0,0 +1,37 @@
+package throttle
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"golang.org/x/time/rate"
+)
+
+type writer struct {
+	limiter *rate.Limiter
+	writer  io.Writer
+}
+
+func NewWriter(w io.Writer, br Byterate) io.Writer {
+	burstLimit := 1024 * 1024 * 1024
+	limiter := rate.NewLimiter(rate.Limit(br), burstLimit)
+	limiter.AllowN(time.Now(), burstLimit)
+	return &writer{
+		limiter: limiter,
+		writer:  w,
+	}
+}
+
+func (w *writer) Write(p []byte) (int, error) {
+	n, err := w.writer.Write(p)
+	if err != nil {
+		return 0, err
+	}
+	r := w.limiter.ReserveN(time.Now(), n)
```

Comment:
Could be worth it separating out `time.Now()` to be a func on the struct. This could make it possible to write tests where we manipulate time.

Reply:
The `time.Now()` passed to Reserve the the origin time for the delay duration, and cant be static for all write calls. Don't really see how this would make testing easier?